apps/src/blockly/customBlocks/googleBlockly/proceduresBlocks.ts

```diff
@@ -183,8 +183,8 @@ GoogleBlockly.Extensions.register(
       flyoutToggleButton,
       renderingInFunctionEditor
     );
-    // Open mini-toolbox by default
-    flyoutToggleButton.setIcon(false);
+    // Open mini-toolbox by default unless the workspace is read-only.
+    flyoutToggleButton.setIcon(this.workspace.options.readOnly);
```

Comment:
Previously we were always opening the mini-toolbox on these definition blocks. This doesn't feel right as toolboxes are normally hidden in read-only mode.